nopslide/firo
# Bip47 lelantus (#1013)

* Add bip47

* Update walletdb.cpp

* Update wallet.h

* Fix Permissions Issue

* Add Space to Trigger Jenkins

* Remove Comments #1

* Remove Extraneous Comments from Recent Payments

* Remove Comments from BIP47 Util

* Remove Extraneous Comments

* Remove Extraneous Comments from the Channel File

* Remove Extraneous Comments from Payment Code Page

* Remove Extraneous Comments in Walletmodel

* Remove Extra Comments

* Remove Extra Comments Related to Old PK Material

* Update addressbookpage.cpp

* Remove Commented Code and Add Comment

* Comment Intentionally Empty if Statement

* Suppress False Flags and Remove Commented Code

* Remove Comments

* Remove Comments and Surpress False Alarms in Wallet and QT Folder

* Remove Comments and LGTM Alerts

* Porting Bip47 to Core14

* remove unused codes

* A little polishing

* remove unused code

* fix error in extracting designated public key in P2PK script

* remove unused function importBip47PaymentChannelData

* fix bugs in address book

* disable delete/edit/new on payment code book

* Fixing linking issues

* Fixing build

* Start BIP47 code refactor
- fix circular dependency issues
- bring tests into one file & remove separate test binary
- correct naming of files, removal of no longer used files

* Class renaming; removing bad code

* refactorings

* merge

* modify rpc commands

* qt bip47 receive screen

* Merge QT

* Merged with master

* Making tests work

* Making tests working

* Minor refactoring, fixing test crashes

* Added the namespace

* Classes were renamed to compy with the convention

* Minor optimization

* Minor refactoring

* First test done

* Minor cleanup

* Ecdh params tests are done

* Done the shared secret test

* CPaymentAddress, CPaymentChannel refactoring

* Address generation based on shared secret is done

* Masked payload has been done

* Minor merge error

* Merge cleanup and minor improvements

* Redesing of bip47 wallet and account has been started

* Account functionality has been split

* Change in comments

* Sending account has been done

* Implemented payment code unmasking

* Preparation for address similarity tests

* Addresses made the same using DH on both sides

* Cleaning up the previous implementation

* Secret addresses in the wallet. Preparatory steps

* Secret addresses added to the wallet.

* Minor improvement for pcode validation

* Minor improvement for address generation

* Preparation for a notification TX sending

* Preparing of a notification tx has been done

* Receiving of notification txs is done

Revert "Receiving of notification txs is done"

This reverts commit 6821b82bab3d10c3852d66bf4feb028c2e6aafd2.

Minor improvements

Minor change

* Minor improvements

* A basic RPC test has been done

* Minor redesign in account numbering

* WIP making notification TX use Lelantus

* Notification tx use Lelantus was done

* Dealing with prelim code review

* Test for Lelantus use

* Serialization work was commenced

* Payment channel serialization tests

* Serialization tests for accounts were done

* Serialization and unit tests added

* Secret privkeys were revised

* Secret privkeys were revised.Minor addition

* Serialization is almost done and tested

* Lelantus fixes

* Reverted the initial code

* Added the RAP form

* Minor improvements

* Fixing Range proof challenge generation

* GUI: Creation of pcodes is done

* Add new challenge generator with CHash256 and use it for new version

* Linking transcripts betweed sigma and balance proofs, using unique_ptr in place of classic pointer

* GUI: Started implementing Sending to pcode

* GUI: Sending to pcode dialog was designed

* GUI: Sending funds to pcode GUI has almost been done.

* GUI: Sending funds to pcode GUI has been done.

* Not using std::any

* Use jsplit version instead of comparing height, add version into domain separators, add pubkeys into transcript

* Opted out from use size_t in address numbering

* Minor fixes

* Adding some comments

* Minor refactoring and bugfixes, Some advanced features are implemented

* Minor fix for test memory leaks

* Removing unneeded functions from SigmaExtendedProver/Verifier

* Review comments applied

* Adding schnorr proof for Qk

* Added check to verify that n-th power is not returning 1

* Use template and merge 2 challenge generator classes

* Review comment applied

* Minor display improvements

* Showing advanced address information

* Allowing bip47 to be null

* Version bump, testnet HF block, testnet blacklist

* Minor improvements, autotests

* QT warnings have been fixed

* Handling Sigma selection for bip47

* Fixing a minor bug

* Version 0.14.6.0

* Gui improvements, bugfixes

* Index loading fixed

* Saving set hash only if there are mints in the block

* Failing test fixed

* Fixing db reading error

* Changing HF block for mainnnet

* Fixing the rescan issues

* Setting mainnet HF block

* Check unique-properties of ProUpRegTx when building new list

* Unlock dialog was added

* Tests have been fixed

* Fixing the lelantus discrepancies

* Code style fix

* Starting the encrypted wallet fix

* The encrypted wallet fix

* Fix for Unlock close crash, cannot send when unconfirmed

* Merge with master

* Version v0.14.7.0

* Improving tests stability

* Adding the Reconsider bip47 transaction menu item

* Transaction description was improved

* Dealing with lgtm comments

* Making the filenames unique

* Making the filenames unique

* Adding Pcodes to the address book

* Fixing the code review comments

* BIP47 pcodes addressbook

* BIP47 pcodes addressbook. Minor improvements

* Improving the OP_RETURN data display

* Improving the tx long descriptiong

* Setting used address number was done

* Resolving Macos build issues

* Minor GUI: balance display, first time message

* GUI improvements: not showing secret address dialog after a notification tx

* GUI: unlock message was done

* GUI: Added hints to send pcode dialog

* GUI: wording change

* GUI: wording change

* Fixing macos build

* GUI: Transaction page now supports pcodes

* Fixing typo and uninitialized value

* GUI: minor fixes and improvements

* GUI: typo fixes

* GUI: minor size fixes

* Locking the notification tx output

* Dealing with code review comments

* Dealing with code review comments

Co-authored-by: Rasikh Morani <rasikh@moranilabs.com>
Co-authored-by: top1st <riguang102@gmail.com>
Co-authored-by: Rasikh Morani <Rasikh@arcadia.agency>
Co-authored-by: Rasikh Morani <rasikh@arcadiamgroup.com>
Co-authored-by: Levon Petrosyan <petrosyan.levon93@gmail.com>
Co-authored-by: Cam <damvananh91@gmail.com>
Co-authored-by: levonpetrosyan93 <45027856+levonpetrosyan93@users.noreply.github.com>
Co-authored-by: riordant <riordant@tcd.ie>
Co-authored-by: Andrey <andrey@zcoin.io>

## PR intention
DeleteMe: Mandatory: What this PR is intended to do, what change it introduces, what issue it solves.

## Code changes brief
DeleteMe: Optional: What are the architectural, UX, or other kind of changes are made which are hard to induce from the code.
